### PR TITLE
Add type hints arelle.ModelObject.py

### PR DESCRIPTION
**Pull request by ggravlingen**
#### Reason for change
Adding more type hints.

#### Description of change
- Doing some cleanup of circular imports
- Adding type hints to `arelle.ModelObject`

#### Steps to Test

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/plugin/validate/ESEF/__init__.py
Comment (ggravlingen) [inline arelle/plugin/validate/ESEF/__init__.py:699]: `id` uses `get`, which by default returns a value and `None` so we need to introduce a guard against `None` here (or don't use get)

Comment (austinmatherne-wk) [reply arelle/plugin/validate/ESEF/__init__.py:699]: You can assert these IDs are not None. [Contexts must have an ID](https://www.xbrl.org/Specification/XBRL-2.1/REC-2003-12-31/XBRL-2.1-REC-2003-12-31+corrected-errata-2013-02-20.html#_4.7.1) and they're coming from `ModelXbrl.contexts` which is a dict of contexts by ID. If they really don't have an ID they would fail to be included in the map these are built from.

File: arelle/ModelObject.py
Comment (austinmatherne-wk) [inline arelle/ModelObject.py:260]: Sorry, my [previous comment](https://github.com/Arelle/Arelle/pull/328#discussion_r967912272) wasn't clear. `ModelContext` has to have an `id`, but there are other `ModelObject` sub classes which don't share that requirement.

Comment (ggravlingen) [reply arelle/ModelObject.py]: Looks like we didn't have to do an assert in `ESEF/__init__.py` when changing the property. At least not locally, let's see what the CI/CD says.

**Review by austinmatherne-wk — APPROVED** [review]

**Comment by ggravlingen** [comment]
@austinmatherne-wk are you waiting on anything from me here? My belief is that this is in que for your internal QA but just wanted to be sure.

**Review by sagesmith-wf — APPROVED** [review]

**Comment by austinmatherne-wk** [comment]
> @austinmatherne-wk are you waiting on anything from me here? My belief is that this is in que for your internal QA but just wanted to be sure.

Nope. You're good. We have a policy of at least two code reviews from the team before QA/merging.

**Comment by ggravlingen** [comment]
> > @austinmatherne-wk are you waiting on anything from me here? My belief is that this is in que for your internal QA but just wanted to be sure.
> 
> 
> 
> Nope. You're good. We have a policy of at least two code reviews from the team before QA/merging. 

Thanks!

**Pull request merged**
